app/src/main/java/com/vci/vectorcamapp/complete_session/list/presentation/components/CompleteSessionListTile.kt

```diff
@@ -216,10 +220,12 @@ fun CompleteSessionListTile(
                             }
                         }
                         Text(
-                            text = if (sessionUploadProgress.totalImageCount == 0) "No images to upload"
-                            else "${sessionUploadProgress.uploadedImageCount} / ${sessionUploadProgress.totalImageCount} images",
-                            style = MaterialTheme.typography.bodySmall,
-                            color = MaterialTheme.colors.textSecondary
+                            text = if (sessionMetadataUploaded && (sessionUploadProgress.totalCount == 0 || (sessionUploadProgress.uploadedMetadataCount == sessionUploadProgress.totalCount && sessionUploadProgress.uploadedImageCount == sessionUploadProgress.totalCount))) "Completed"
```

Comment:
I think you can remove the branch which says sessionUploadProgress.totalCount == 0